gocourse05/zooobservation/camera/camera.go

```diff
@@ -0,0 +1,90 @@
+package camera
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sarff/prjctr-golang_beginning/gocourse05/zooobservation/animal"
+)
+
+type Direction string
+
+const (
+	Left   Direction = "Left"
+	Right  Direction = "Right"
+	Top    Direction = "Top"
+	Bottom Direction = "Bottom"
+)
+
+type HistoryItem struct {
+	Time      time.Time
+	Direction Direction
+	ID        int
+}
+
+type Camera interface {
+	DetectMovement(direction Direction, historyItems []HistoryItem, animalID int) ([]HistoryItem, error)
+	SaveToServer(historyItems []HistoryItem) error
+}
+
+type DayCamera struct{}
+
+func (d *DayCamera) DetectMovement(direction Direction, historyItems []HistoryItem, animalID int) ([]HistoryItem, error) {
+	historyItems = moveToFront(direction, historyItems, animalID)
+	return historyItems, d.SaveToServer(historyItems)
+}
+
+func (d *DayCamera) SaveToServer(historyItems []HistoryItem) error {
+	fmt.Println("Simulation: DayCamera history saved:", historyItems)
+	return nil
+}
+
+type NightCamera struct{}
+
+func (n *NightCamera) DetectMovement(direction Direction, historyItems []HistoryItem, animalID int) ([]HistoryItem, error) {
+	historyItems = moveToFront(direction, historyItems, animalID)
+	return historyItems, n.SaveToServer(historyItems)
+}
+
+func (n *NightCamera) SaveToServer(historyItems []HistoryItem) error {
+	fmt.Println("Simulation: NightCamera history saved:", historyItems)
+	return nil
+}
+
+type Controller struct {
+	DayCamera   DayCamera
+	NightCamera NightCamera
+}
+
+func (c *Controller) Move(animal animal.Animal, direction Direction, historyItems []HistoryItem) ([]HistoryItem, error) {
+	var camera Camera
+	camera = &c.NightCamera
```

Comment:
Здається, так має працювати:
```suggestion
	camera := Camera(&c.nightCamera)
```